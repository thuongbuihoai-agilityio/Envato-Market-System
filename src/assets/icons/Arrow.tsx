// Types
import { TColor } from '@interfaces/index';

export const Arrow = ({ color, rotate }: TColor) => (
  <svg
    stroke={color}
    width="24"
    height="24"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    style={{
      transform: `rotate(${rotate})`,
    }}
  >
    <path
      d="M7 10L12 14L17 10"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    ></path>
  </svg>
);
