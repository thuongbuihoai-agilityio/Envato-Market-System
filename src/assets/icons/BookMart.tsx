export const BookMark = () => (
  <svg width="24" height="25" viewBox="0 0 24 25" stroke="#a0aec0" fill="none">
    <path
      d="M19 21.5659L12 16.5659L5 21.5659V5.56592C5 5.03548 5.21071 4.52678 5.58579 4.1517C5.96086 3.77663 6.46957 3.56592 7 3.56592H17C17.5304 3.56592 18.0391 3.77663 18.4142 4.1517C18.7893 4.52678 19 5.03548 19 5.56592V21.5659Z"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    ></path>
  </svg>
);
