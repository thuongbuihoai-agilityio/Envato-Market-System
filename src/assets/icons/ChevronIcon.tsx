export const ChevronIcon = () => (
  <svg
    width="21"
    height="21"
    viewBox="0 0 21 21"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    style={{ transform: 'translateY(4px)' }}
  >
    <path
      d="M7.72168 5.03271L12.7217 10.0327L17.7217 5.03271"
      stroke="#A0AEC0"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    ></path>
  </svg>
);
