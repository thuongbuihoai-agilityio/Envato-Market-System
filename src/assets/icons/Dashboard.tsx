export const DashboardIcon = () => (
  <svg
    width="18"
    height="21"
    viewBox="0 0 18 21"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="path-1"
      d="M0 8.84719C0 7.99027 0.366443 7.17426 1.00691 6.60496L6.34255 1.86217C7.85809 0.515019 10.1419 0.515019 11.6575 1.86217L16.9931 6.60496C17.6336 7.17426 18 7.99027 18 8.84719V17C18 19.2091 16.2091 21 14 21H4C1.79086 21 0 19.2091 0 17V8.84719Z"
      fill="#1A202C"
    ></path>
    <path
      className="path-2"
      d="M5 17C5 14.7909 6.79086 13 9 13C11.2091 13 13 14.7909 13 17V21H5V17Z"
      fill="#22C55E"
    ></path>
  </svg>
);
