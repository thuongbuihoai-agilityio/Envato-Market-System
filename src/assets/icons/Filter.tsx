import { SVGProps } from 'react';

export const FilterIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width={18}
    height={17}
    fill="none"
    {...props}
  >
    <path d="M7.552 13.502H1.25M10.362 3.81h6.301" />
    <path
      d="M5.948 3.756a2.355 2.355 0 0 0-2.363-2.347 2.355 2.355 0 0 0-2.363 2.347 2.355 2.355 0 0 0 2.363 2.346 2.355 2.355 0 0 0 2.363-2.346ZM17.221 13.463a2.354 2.354 0 0 0-2.362-2.346 2.355 2.355 0 0 0-2.364 2.346 2.355 2.355 0 0 0 2.364 2.347 2.354 2.354 0 0 0 2.362-2.347Z"
      clipRule="evenodd"
    />
  </svg>
);
