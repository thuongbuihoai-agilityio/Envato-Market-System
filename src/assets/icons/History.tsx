export const HistoryIcon = () => (
  <svg
    width="18"
    height="21"
    viewBox="0 0 18 21"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17.5 12.5C17.5 17.1944 13.6944 21 9 21C4.30558 21 0.5 17.1944 0.5 12.5C0.5 7.80558 4.30558 4 9 4C13.6944 4 17.5 7.80558 17.5 12.5Z"
      fill="#1A202C"
      className="path-1"
    ></path>
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M8.99995 1.75C8.02962 1.75 7.09197 1.88462 6.20407 2.13575C5.80549 2.24849 5.39099 2.01676 5.27826 1.61818C5.16553 1.21961 5.39725 0.805108 5.79583 0.692376C6.81525 0.404046 7.89023 0.25 8.99995 0.25C10.1097 0.25 11.1846 0.404046 12.2041 0.692376C12.6026 0.805108 12.8344 1.21961 12.7216 1.61818C12.6089 2.01676 12.1944 2.24849 11.7958 2.13575C10.9079 1.88462 9.97028 1.75 8.99995 1.75Z"
      fill="#22C55E"
      className="path-2"
    ></path>
    <path
      d="M11 13C11 14.1046 10.1046 15 9 15C7.89543 15 7 14.1046 7 13C7 11.8954 7.89543 11 9 11C10.1046 11 11 11.8954 11 13Z"
      fill="#22C55E"
      className="path-2"
    ></path>
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M9 7.25C9.41421 7.25 9.75 7.58579 9.75 8V12C9.75 12.4142 9.41421 12.75 9 12.75C8.58579 12.75 8.25 12.4142 8.25 12V8C8.25 7.58579 8.58579 7.25 9 7.25Z"
      fill="#22C55E"
      className="path-2"
    ></path>
  </svg>
);
