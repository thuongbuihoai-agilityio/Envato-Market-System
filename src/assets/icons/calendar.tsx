export const CalendarIcon = () => (
  <svg
    width="18"
    height="21"
    viewBox="0 0 18 21"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 6.5C0 4.29086 1.79086 2.5 4 2.5H14C16.2091 2.5 18 4.29086 18 6.5V8V17C18 19.2091 16.2091 21 14 21H4C1.79086 21 0 19.2091 0 17V8V6.5Z"
      fill="#1A202C"
      className="path-1"
    ></path>
    <path
      d="M14 2.5H4C1.79086 2.5 0 4.29086 0 6.5V8H18V6.5C18 4.29086 16.2091 2.5 14 2.5Z"
      fill="#22C55E"
      className="path-2"
    ></path>
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M5 0.25C5.41421 0.25 5.75 0.585786 5.75 1V4C5.75 4.41421 5.41421 4.75 5 4.75C4.58579 4.75 4.25 4.41421 4.25 4V1C4.25 0.585786 4.58579 0.25 5 0.25ZM13 0.25C13.4142 0.25 13.75 0.585786 13.75 1V4C13.75 4.41421 13.4142 4.75 13 4.75C12.5858 4.75 12.25 4.41421 12.25 4V1C12.25 0.585786 12.5858 0.25 13 0.25Z"
      fill="#1A202C"
      className="path-2"
    ></path>
    <circle cx="9" cy="14" r="1" fill="#22C55E"></circle>
    <circle cx="13" cy="14" r="1" fill="#22C55E" className="path-2"></circle>
    <circle cx="5" cy="14" r="1" fill="#22C55E" className="path-2"></circle>
  </svg>
);
