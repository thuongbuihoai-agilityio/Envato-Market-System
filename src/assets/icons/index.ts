export * from './Bell';
export * from './Email';
export * from './Gift';
export * from './Theme';
export * from './User';
export * from './UserMini';
export * from './LogOut';
export * from './Arrow';
export * from './Dashboard';
export * from './Transaction';
export * from './Statistic';
export * from './Analytic';
export * from './MyWallet';
export * from './Inbox';
export * from './Integration';
export * from './Calendar';
export * from './History';
export * from './Support';
export * from './Setting';
export * from './ComingSoon';
export * from './NotFound';
export * from './SignOut';
export * from './Rotate';
export * from './Search';
export * from './sort';
export * from './Account';
