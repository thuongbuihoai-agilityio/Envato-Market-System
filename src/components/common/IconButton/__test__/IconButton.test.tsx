import { Theme } from '@assets/icons';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// component
import { IconButton } from '@components/index';

describe('IconButton test cases', () => {
  const mockOnClick = jest.fn();
  it('should render correctly', () => {
    const { container } = render(
      <IconButton onClick={mockOnClick}>
        <Theme colorFill="gray.800" />
      </IconButton>,
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onClick when being clicked', async () => {
    const { getByTestId } = render(
      <IconButton onClick={mockOnClick}>
        <Theme colorFill="gray.800" />
      </IconButton>,
    );
    const changeTheme = getByTestId('icon-button');

    await userEvent.click(changeTheme);

    expect(mockOnClick).toHaveBeenCalled();
  });

  it('render with hasNewNotification', async () => {
    const { getByTestId } = render(
      <IconButton hasNewNotification onClick={mockOnClick}>
        <Theme colorFill="gray.800" />
      </IconButton>,
    );
    const iconButton = getByTestId('icon-button-component');
    expect(iconButton).toBeInTheDocument();
  });

  it('render with isNotification with hasNewNotification', async () => {
    const { getByTestId } = render(
      <IconButton isNotification hasNewNotification onClick={mockOnClick}>
        <Theme colorFill="gray.800" />
      </IconButton>,
    );
    const iconButton = getByTestId('icon-button-component');
    expect(iconButton).toBeInTheDocument();
  });

  it('render with isNotification without hasNewNotification', async () => {
    const { getByTestId } = render(
      <IconButton isNotification onClick={mockOnClick}>
        <Theme colorFill="gray.800" />
      </IconButton>,
    );
    const iconButton = getByTestId('icon-button-component');
    expect(iconButton).toBeInTheDocument();
  });

  it('render without onClick', async () => {
    const { getByTestId } = render(
      <IconButton>
        <Theme colorFill="gray.800" />
      </IconButton>,
    );
    const iconButton = getByTestId('icon-button-component');
    expect(iconButton).toBeInTheDocument();
  });
});
