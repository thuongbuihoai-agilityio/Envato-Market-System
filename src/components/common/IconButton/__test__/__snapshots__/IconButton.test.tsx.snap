// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton test cases should render correctly 1`] = `
<div>
  <div
    class="css-1ve4u7y"
    data-testid="icon-button-component"
  >
    <button
      aria-label="Send email"
      class="chakra-button css-yt1s1n"
      data-testid="icon-button"
      type="button"
    >
      <svg
        fill="none"
        height="24"
        stroke="gray.800"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.3284 14.8687C13.249 14.8687 9.13135 10.751 9.13135 5.67163C9.13135 4.74246 9.26914 3.84548 9.5254 3C5.74897 4.14461 3 7.65276 3 11.803C3 16.8824 7.11765 21 12.197 21C16.3472 21 19.8554 18.251 21 14.4746C20.1545 14.7309 19.2575 14.8687 18.3284 14.8687Z"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </button>
  </div>
</div>
`;
